### DISCUSSION TITLE
LittleFS data upload - running from outside the IDE

### DISCUSSION THREAD

**Discussion by obdevel** [earlephilhower/arduino-pico#1477] [Q&A] [answered]
I'd like to provide my users with a means to upload a data folder to an existing Pico. My search skills are failing (and the new GitHub UI is fighting me). What is the minimum required (other than Python) and is there an example command line ?

Thanks.

**Comment by earlephilhower** [comment]
What I would suggest is doing one upload with upload verbosity (File->Preferences->Show Verbose Output->Upload) enabled.  I believe you get all the command line options needed to do the upload in your log, which you can then cut and paste in your own scripts.

**Comment by obdevel** [comment]
Thanks. I already had that switched on and tried to grep through the source for likely strings. This is all I see on the output:
```
[LittleFS] data    : /Users/xxx/Documents/Arduino/Signal_box_v1/data
[LittleFS] size    : 256KB
[LittleFS] page    : 256
[LittleFS] block   : 4096
/config holt.txt
/config.txt
/merg-logo-48.png
/NotoSansBold15.vlw
/bell1.wav
/NotoSansBold36.vlw
/config miss.txt
skipping .DS_Store
[LittleFS] upload   : /var/folders/03/s3pv8qk91w78m4_st_nbxl5m0000gn/T/arduino_build_672716/Signal_box_v1.mklittlefs.bin
[LittleFS] address  : 270266368
[LittleFS] swerial  : /dev/cu.usbmodem142201
[LittleFS] python   : /Users/xxx/Library/Arduino15/packages/rp2040/tools/pqt-python3/1.0.1-base-3a57aed/python3
[LittleFS] uploader : /Users/xxx/Library/Arduino15/packages/rp2040/hardware/rp2040/3.2.1/tools/uf2conv.py

Resetting /dev/cu.usbmodem142201
Converting to uf2, output size: 524288, start address: 0x101bf000
Scanning for RP2040 devices
Flashing /Volumes/RPI-RP2 (RPI-RP2)
Wrote 524288 bytes to /Volumes/RPI-RP2/NEW.UF2
```

I think I'm fine with uf2.conv.py for the actual uploading, but I don't know how to create the .bin file from the filesystem data directory. I'll try following the path from clicking on that menu item.

**Reply by earlephilhower** [reply, marked as answer]
Ah, I guess I don't print out the actual command line.  I think you have all the info needed, though.

Just run mklittlefs (installed as part of the toolchain, or downloadable from GH) to make the FS image.  You'll need to turn "256k" into the byte count, but otherwise it's cut-n-paste:

````
mklittlefs -c dir-to-pack -b 4096 -p 256 -s 262144 out.bin
````

**Comment by obdevel** [comment]
That's it. Thanks.

Seems I only have the macOS binary on my system (and it's also installed with the esp32 and esp8266 cores). So I assume the Boards Manager is intelligent enough to only download and stage what is required, and therefore a Windows user will have a Windows binary, etc.
